crates/polars-io/src/csv/read/reader.rs

```diff
@@ -161,70 +157,63 @@ impl<R: MmapBytesReader> CsvReader<R> {
     // * Move this step outside of the reader so that we don't do it multiple times
     //   when we read a file list.
     // * See if we can avoid constructing a filtered schema.
-    fn prepare_schema_overwrite(
-        &self,
-        overwriting_schema: &Schema,
-    ) -> PolarsResult<(Schema, Vec<Field>, bool)> {
+    fn prepare_schema(&mut self) -> PolarsResult<bool> {
         // This branch we check if there are dtypes we cannot parse.
         // We only support a few dtypes in the parser and later cast to the required dtype
-        let mut to_cast = Vec::with_capacity(overwriting_schema.len());
-
         let mut _has_categorical = false;
-        let mut _err: Option<PolarsError> = None;
-
-        #[allow(unused_mut)]
-        let schema = overwriting_schema
-            .iter_fields()
-            .filter_map(|mut fld| {
-                use DataType::*;
-                match fld.data_type() {
-                    Time => {
-                        to_cast.push(fld);
-                        // let inference decide the column type
-                        None
-                    },
-                    #[cfg(feature = "dtype-categorical")]
-                    Categorical(_, _) => {
-                        _has_categorical = true;
-                        Some(fld)
-                    },
-                    #[cfg(feature = "dtype-decimal")]
-                    Decimal(precision, scale) => match (precision, scale) {
-                        (_, Some(_)) => {
-                            to_cast.push(fld.clone());
+
+        let mut process_schema = |schema: &Schema| {
+            schema
+                .iter_fields()
+                .map(|mut fld| {
+                    use DataType::*;
+
+                    match fld.data_type() {
+                        Time => {
+                            self.options.fields_to_cast.push(fld.clone());
+                            // let inference decide the column type
```

Comment:
This comment isn't correct if set it to `string` and return Some`.

Reply:
Good catch. To clarify is setting to string acceptable here? Returning `None` can cause bugs for `schema` because it leads to an error `at least 1 column must be projected` if all the columns are times.